external-crates/move/move-compiler/src/naming/translate.rs

```diff
@@ -1308,8 +1335,19 @@ fn exp_(context: &mut Context, e: E::Exp) -> N::Exp {
         EE::IfElse(eb, et, ef) => {
             NE::IfElse(exp(context, *eb), exp(context, *et), exp(context, *ef))
         }
-        EE::While(eb, el) => NE::While(exp(context, *eb), exp(context, *el)),
-        EE::Loop(el) => NE::Loop(exp(context, *el)),
+        EE::While(eb, el) => {
+            let cond = exp(context, *eb);
+            let loop_name = context.enter_loop(eloc);
```

Comment:
Idea, to ever so slightly decrease the stack size on the recursive call `exp`, we could have `context.exit_loop` give back the name? 

(I think you should find it highly comical that I'm giving advice on this topic. Our stack usage in the compiler is off the charts lol)